webpush/utils.py

```diff
@@ -2,26 +2,62 @@
 from django.forms.models import model_to_dict
 from django.urls import reverse
 
-from pywebpush import webpush
+from pywebpush import WebPushException, webpush
 
 
 def send_notification_to_user(user, payload, ttl=0):
     # Get all the push_info of the user
+
+    errors = []
     push_infos = user.webpush_info.select_related("subscription")
     for push_info in push_infos:
-        _send_notification(push_info, payload, ttl)
+        try:
+            _send_notification(push_info.subscription, payload, ttl)
+
+        except WebPushException as ex:
+            errors.append(dict(subscription=push_info.subscription,
+                               exception=ex))
+
+    if errors:
+        raise WebPushException("Push failed.", extra=errors)
 
 
 def send_notification_to_group(group_name, payload, ttl=0):
     from .models import Group
     # Get all the subscription related to the group
+
+    errors = []
     push_infos = Group.objects.get(name=group_name).webpush_info.select_related("subscription")
     for push_info in push_infos:
-        _send_notification(push_info, payload, ttl)
+        try:
+            _send_notification(push_info.subscription, payload, ttl)
+
+        except WebPushException as ex:
+            errors.append(dict(subscription=push_info.subscription,
+                               exception=ex))
+
+    if errors:
+        raise WebPushException("Push failed.", extra=errors)
+
+
+def send_to_subscription(subscription, payload, ttl):
+    _send_notification(subscription, payload, ttl)
+
+
+def send_to_subscriptions(queryset, payload, ttl):
```

Comment:
I think `send_to_subscription`  should be implemented. There are very less use case of it and it can be achieved by `send_to_subscription` method alaone.